sdk/storage/Azure.Storage.Blobs.Batching/src/Multipart.cs

```diff
@@ -0,0 +1,257 @@
+                    response.AddHeader(name, value);
+
+                    line = await body.ReadLineAsync(async, cancellationToken).ConfigureAwait(false);
+                }
+
+                // Copy the rest of the body as the response content
+                var responseContent = new MemoryStream();
+                if (async)
+                {
+                    await body.CopyToAsync(responseContent).ConfigureAwait(false);
+                }
+                else
+                {
+                    body.CopyTo(responseContent);
+                }
+                responseContent.Seek(0, SeekOrigin.Begin);
+                response.ContentStream = responseContent;
+            }
+
+            // Collect the responses and order by Content-ID
+            Response[] ordered = new Response[responses.Count];
+            for (int i = 0; i < ordered.Length; i++)
+            {
+                ordered[i] = responses[i];
+            }
+            return ordered;
+        }
+
+        /// <summary>
+        /// Read the next line of text.
+        /// </summary>
+        /// <param name="stream">The stream to read from.</param>
+        /// <param name="async">
+        /// Whether to invoke the operation asynchronously.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate notifications
+        /// that the operation should be cancelled.
+        /// </param>
+        /// <returns>The next line of text.</returns>
+        internal static async Task<string> ReadLineAsync(
+            this BufferedReadStream stream,
+            bool async,
+            CancellationToken cancellationToken) =>
+            async ?
+                await stream.ReadLineAsync(BatchConstants.ResponseLineSize, cancellationToken).ConfigureAwait(false) :
+                stream.ReadLine(BatchConstants.ResponseLineSize);
+
+        /// <summary>
+        /// Read the next multipart section
+        /// </summary>
+        /// <param name="reader">The reader to parse with.</param>
+        /// <param name="async">
+        /// Whether to invoke the operation asynchronously.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate notifications
+        /// that the operation should be cancelled.
+        /// </param>
+        /// <returns>The next multipart section.</returns>
+        internal static async Task<MultipartSection> GetNextSectionAsync(
+            this MultipartReader reader,
+            bool async,
+            CancellationToken cancellationToken) =>
+            async ?
+                await reader.ReadNextSectionAsync(cancellationToken).ConfigureAwait(false) :
+                reader.ReadNextSectionAsync(cancellationToken).GetAwaiter().GetResult(); // TODO: Decide if we need a proper sync API here
```

Comment:
Do we need to resolve the TODO?

Reply:
No, that was waiting for a bug.  Filed #7972 and updated.